src/pal/src/thread/process.cpp

```diff
@@ -4313,43 +4310,47 @@ getPath(
     if (lpwstr)
     {
         /* convert path to multibyte, check buffer size */
-        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, lpPathFileName, iLen,
+        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, NULL, 0,
             NULL, NULL);
-        if (n == 0)
+        
+        if (!lpPathFileName.Reserve(n + lpFileNameString.GetCount() ))
```

Comment:
I am still adding a 100 characters on every resize on the heap, so these smaller appends should be fine